frb_codegen/src/ir/ty_type_impl_trait.rs

```diff
@@ -0,0 +1,86 @@
+use crate::ir::*;
+use crate::target::Target;
+use convert_case::{Case, Casing};
+use syn::Ident;
+
+#[derive(Debug, Clone, Hash, Eq, PartialEq)]
+pub struct IrTypeImplTrait {
```

Comment:
Ok I see the problem: "Ir" is "intermediate representation" and means we should write down code generator for both Dart and Rust. If you want `IrTypeImplTrait` to generate an enum, maybe consider the special `ir_delegate`. For example, the underlying implementation for String is indeed `Vec<u8>`, so you can see how we do it in ir_delegate files